store/src/migrations.rs

```diff
@@ -5,24 +5,59 @@ pub static MIGRATIONS: Lazy<Migrations> = Lazy::new(|| {
     Migrations::new(vec![M::up(
         "
         CREATE TABLE
-            block_header
+            block_headers
         (
             block_num INTEGER NOT NULL,
             block_header BLOB NOT NULL,
 
             PRIMARY KEY (block_num),
-            CONSTRAINT block_header_block_num_positive CHECK (block_num >= 0)
+            CONSTRAINT block_header_block_num_is_u32 CHECK (block_num >= 0 AND block_num < 4294967296)
```

Comment:
The previous store PR requested for the usage of `u32` as the block number. This PR does some tidying up, to add that to the DB constraints and type system.